File: app/src/main/java/edu/harding/pokemonteambuilder/Move.java
package edu.harding.pokemonteambuilder;

public class Move{
    public Move(String name, Boolean effect, Boolean physical,
                String flavorText, int power, int accuracy){
        mName = name;
        mEffect = effect;
        mPhysical = physical;
        mFlavorText = flavorText;
        mPower = power;
        mAccuracy = accuracy;
    }

    private String mName;
    private Boolean mEffect;
    private Boolean mPhysical;
    private String mFlavorText;
    private int mPower;
    private int mAccuracy;
}

File: app/src/main/java/edu/harding/pokemonteambuilder/Move.kt
package edu.harding.pokemonteambuilder

class Move {

    var name: String? = null
    var effect: Boolean? = null
    var physical: Boolean? = null
    var flavorText: String? = null
    var power: Int = 0
    var accuracy: Int = 0
    var learnLevel: Int = 0

    constructor(name: String, flavorText: String, accuracy: Int) {
        this.name = name
        effect = true
        physical = null
        this.flavorText = flavorText
        power = 0
        this.accuracy = accuracy
    }

    constructor(name: String, physical: Boolean?, flavorText: String, power: Int, accuracy: Int) {
        this.name = name
        effect = false
        this.physical = physical
        this.flavorText = flavorText
        this.power = power
        this.accuracy = accuracy
    }

    constructor(name: String, flavorText: String, accuracy: Int, learnLevel: Int) {
        this.name = name
        effect = true
        physical = null
        this.flavorText = flavorText
        power = 0
        this.accuracy = accuracy
        this.learnLevel = learnLevel
    }

    constructor(name: String, physical: Boolean?, flavorText: String,
                power: Int, accuracy: Int, learnLevel: Int) {
        this.name = name
        effect = false
        this.physical = physical
        this.flavorText = flavorText
        this.power = power
        this.accuracy = accuracy
        this.learnLevel = learnLevel
    }


}
